src/dataflow-types/src/types/sources.proto

```diff
@@ -48,6 +48,13 @@ message ProtoPersistSourceConnector {
     string shard_id = 3;
 }
 
+message ProtoS3KeySource {
+    oneof kind {
+        string scan = 1;
+        string sqs_notifications = 2;
+    }
```

Comment:
if someone in the future adds a new field to the `Scan` variant on the enum, will they need to add a new kind here with a sub-message?

Reply:
Yes! My assumption here was that it seemed unlikely that we would just another field to `Scan`, but if you think otherwise, I can make a separate sub-message for it right now.